posthog/temporal/workflows/squash_person_overrides.py

```diff
@@ -0,0 +1,256 @@
+import itertools
+import operator
+from collections.abc import Iterator
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta
+from uuid import UUID
+
+import psycopg2
+from django.conf import settings
+from temporalio import activity, workflow
+
+from posthog.clickhouse.client.execute import sync_execute
+
+SELECT_PERSONS_TO_OVERRIDE_QUERY = """
+SELECT
+    team_id,
+    old_person_id
+FROM
+    {database}.person_overrides
+WHERE
+    _partition_id = %(partition_id)s;
```

Comment:
Latest commit tried to ensure we are not deleting persons that have overrides in the past. Your comment refers to ensuring we are not deleting persons that have overrides "in the future" (after the workflow starts). Luckily there are no more time dimensions we need to worry about after this one :sweat_smile:

The heuristic could work, but I'm thinking if we could just include `merged_at` when deleting so that we only delete stuff that was present before or at `merged_at`, ignoring any merges that happened after.

Reply:
The disadvantage of the heuristic is that data from PG could be delayed longer than an hour in the presence of heavy lag. So if possible I'd prefer a more reliable solution.